src/lib/Persistence/Legacy/ContentModelGateway.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\FieldTypeRichText\Persistence\Legacy;
+
+use Doctrine\DBAL\Connection;
+
+class ContentModelGateway
+{
+    public const DB_TABLE_CONTENTOBJECT_ATTRIBUTE = 'ezcontentobject_attribute';
+    public const FIELD_TYPE_IDENTIFIER = 'ezrichtext';
+
+    /**
+     * @var \Doctrine\DBAL\Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function countRichtextAttributes(): int
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query->select('count(distinct a.id)')
+            ->from(self::DB_TABLE_CONTENTOBJECT_ATTRIBUTE, 'a')
+            ->where(
+                $query->expr()->eq(
+                    'a.data_type_string',
+                    ':data_type'
+                )
+            )
+            ->setParameter(':data_type', self::FIELD_TYPE_IDENTIFIER);
+
+        $statement = $query->execute();
+
+        return (int) $statement->fetchOne();
```

Comment:
What exactly are you trying to count here?

Reply:
Counting number of rows in contentobject_attribute with unique `id`s. 

This is then used for for the progressbar so that it can show any usefull information

Reply:
Reason why I want that number, and not the total number of rows is that I do not want to use the slow `offset` when iterating over the table. Thus I generate a cursor based in `id` and we then need to know how many different values of `id` we are dealing with in order for the progressbar to have any value...